crates/modelardb_compression/src/models/swing.rs

```diff
@@ -326,6 +350,26 @@ fn compute_slope_and_intercept(
     }
 }
 
+/// Computes numerator and denominator for Equation 6 
+/// in Section 3.2 of [Swing and Slide paper](https://dl.acm.org/doi/10.14778/1687627.1687645).
+/// It is a part of recording mechanism for finding the optimal slope with minimum MSE.
+fn compute_local_optimal_slope(
```

Comment:
Unclear what `local` and `optimal` means in this context, optimal can almost always be replaced with a more precise term. For example, in this case is `optimal` the lowest or highest value and for what metric? Maybe rename the function to more clearly state what it computes.